src/main/java/dev/jbang/net/JdkManager.java

```diff
@@ -203,6 +207,40 @@ public static void uninstallJdk(int version) {
 		}
 	}
 
+	/**
+	 * Links JBang JDK folder to an already existing JDK path with a soft link. It
+	 * checks if the incoming version number is the same that the linked JDK has, if
+	 * not an exception will be raised.
+	 *
+	 * @param path    path to the pre-installed JDK.
+	 * @param version requested version to link.
+	 */
+	public static void linkToExistingDirectory(String path, int version) {
+		Path jdkPath = getJdkPath(version);
+		Util.verboseMsg("Trying to link " + path + " to " + jdkPath);
+		if (Files.exists(jdkPath)) {
+			Util.verboseMsg("JBang managed JDK already exists, must be deleted to make sure linking works");
+			try {
+				FileUtils.deleteDirectory(jdkPath.toFile());
+			} catch (IOException e) {
+				throw new ExitException(EXIT_UNEXPECTED_STATE,
+						"Unable delete already existing JBang managed JDK at " + jdkPath, e);
+			}
+		}
+		Path linkedJdkPath = Paths.get(path);
+		if (!Files.isDirectory(linkedJdkPath)) {
+			throw new ExitException(EXIT_INVALID_INPUT, "Unable to resolve path as directory: " + path);
+		}
+		Optional<Integer> ver = resolveJavaVersionFromPath(linkedJdkPath);
+		if (ver.isPresent() && ver.get() == version) {
+			Util.createLink(jdkPath, linkedJdkPath);
+			Util.infoMsg("JDK " + version + " has been linked to: " + linkedJdkPath);
+		} else {
+			throw new ExitException(EXIT_INVALID_INPUT, "Java version in given path: " + path
+					+ " is not having the requested version " + version + ", please provide a proper path");
```

Comment:
It may not be infer the X value, cause this path could point to anywhere, it can have no Java inside it, I can do an evaluation and if it does not work then fallback to the original message if it is okay.